### ISSUE TITLE
informative pair not complete on backtesting 

### ISSUE THREAD

**Issue by M-t-t-h** [freqtrade/freqtrade#8974]
## Describe your environment
Operating system: Docker on Synology
Freqtrade Version: Latest with plots

## Describe the problem:
On populate_indicators function i've got:
```
inf_tf = '4h'
 informative = self.dp.get_pair_dataframe(pair=metadata['pair'], timeframe=inf_tf)
        informative['date'] = dataframe['date']        
        informative['close'] = dataframe['close']        
        informative['rsi'] = ta.RSI(informative)
        infoMacD = ta.MACD(informative)
        informative['macd'] = infoMacD['macd']
        informative['macdsignal'] = infoMacD['macdsignal']
        informative['macdhist'] = infoMacD['macdhist']
        # informative['macd'] = macd['macd'](informative, timeperiod=14)
        dataframe = merge_informative_pair(dataframe, informative, self.timeframe, inf_tf, ffill=True) 
```
When play this strategy on backtesting, it's work (but with no much trends by pairs...)
I export plots to get information and i found problem with informative pais.
Only firsts candles are populates (cfr screenshot : macd_4h).

### Steps to reproduce:

  1. populate informative with 4h timeframe (for a strategy on 1h)
  2. run backtesting
  3. plot it and open html
  
### Observed Results:
Informative paris stop to be refreshed after some time
![tmp_ft_screenshot](https://github.com/freqtrade/freqtrade/assets/8056358/4c2eb4ed-8dc2-40bc-9481-fbb1920d96bc)

**Comment by xmatthias** [comment]
Please get used to filling out the issue template.  Those who frequently omit the template may have their issues closed without comment.
We're asking some very basic questions - by deleting the issue template, you're explicitly choosing not to answer our questions.

Filling out the template helps reduce our workload, and we greatly appreciate that.

---

please provide the full logs from the run you did.

**Comment by M-t-t-h** [comment]
Sorry for this missing information (tried to understand all of it but english not my primary langage).

Please find logs of run with the result below (new backtesting, new plots).

What i see it's more confusing than i expected...
RSI_4H and MACD_4H are same as RSI and MACD on graph until it's flat.
![Screenshot_2b](https://github.com/freqtrade/freqtrade/assets/8056358/366820a4-7deb-47bd-9e07-d08f83299b84)

But on print dataframe, looks ok (cfr screenshot 1)

![Screenshot_2A](https://github.com/freqtrade/freqtrade/assets/8056358/d78163f4-268c-45c8-bdb7-64021c2c695f)

Logs :

```
Creating freqtrade_freqtrade_run ... done
2023-07-27 19:26:32,814 - freqtrade - INFO - freqtrade 2023.6
2023-07-27 19:26:32,895 - freqtrade.configuration.load_config - INFO - Using config: user_data/config.json ...
2023-07-27 19:26:32,897 - freqtrade.loggers - INFO - Verbosity set to 0
2023-07-27 19:26:32,897 - freqtrade.configuration.configuration - INFO - Using max_open_trades: 3 ...
2023-07-27 19:26:33,215 - freqtrade.configuration.configuration - INFO - Using user-data directory: /freqtrade/user_data ...
2023-07-27 19:26:33,216 - freqtrade.configuration.configuration - INFO - Using data directory: /freqtrade/user_data/data/kraken ...
2023-07-27 19:26:33,217 - freqtrade.configuration.configuration - INFO - Parameter --cache=day detected ...
2023-07-27 19:26:33,221 - freqtrade.exchange.check_exchange - INFO - Checking exchange...
2023-07-27 19:26:33,230 - freqtrade.exchange.check_exchange - INFO - Exchange "kraken" is officially supported by the Freqtrade development team.
2023-07-27 19:26:33,231 - freqtrade.configuration.configuration - INFO - Using pairlist from configuration.
2023-07-27 19:26:33,231 - freqtrade.configuration.config_validation - INFO - Validating configuration ...
2023-07-27 19:26:33,281 - freqtrade.commands.optimize_commands - INFO - Starting freqtrade in Backtesting mode
2023-07-27 19:26:33,302 - freqtrade.exchange.exchange - INFO - Instance is running with dry_run enabled
2023-07-27 19:26:33,302 - freqtrade.exchange.exchange - INFO - Using CCXT 3.1.44
2023-07-27 19:26:33,303 - freqtrade.exchange.exchange - INFO - Applying additional ccxt config: {'enableRateLimit': True}
2023-07-27 19:26:33,312 - freqtrade.exchange.exchange - INFO - Applying additional ccxt config: {'enableRateLimit': True, 'rateLimit': 3100}
2023-07-27 19:26:33,322 - freqtrade.exchange.exchange - INFO - Using Exchange "Kraken"
2023-07-27 19:26:38,495 - freqtrade.resolvers.exchange_resolver - INFO - Using resolved exchange 'Kraken'...
2023-07-27 19:26:39,037 - freqtrade.resolvers.iresolver - INFO - Using resolved strategy SmallStraV3 from '/freqtrade/user_data/strategies/SmallStraV3.py'...
2023-07-27 19:26:39,037 - freqtrade.strategy.hyper - INFO - Found no parameter file.
2023-07-27 19:26:39,038 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'stake_currency' with value in config file: EUR.
2023-07-27 19:26:39,039 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'stake_amount' with value in config file: unlimited.
2023-07-27 19:26:39,039 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'unfilledtimeout' with value in config file: {'entry': 10, 'exit': 10, 'exit_timeout_count': 0, 'unit': 'minutes'}.
2023-07-27 19:26:39,039 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'max_open_trades' with value in config file: 3.
2023-07-27 19:26:39,039 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using minimal_roi: {'0': 100}
2023-07-27 19:26:39,039 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using timeframe: 1h
2023-07-27 19:26:39,039 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stoploss: -0.03
2023-07-27 19:26:39,040 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop: False
2023-07-27 19:26:39,040 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop_positive: 0.025
2023-07-27 19:26:39,040 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop_positive_offset: 0.03
2023-07-27 19:26:39,040 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_only_offset_is_reached: True
2023-07-27 19:26:39,040 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using use_custom_stoploss: False
2023-07-27 19:26:39,041 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using process_only_new_candles: True
2023-07-27 19:26:39,041 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using order_types: {'entry': 'limit', 'exit': 'limit', 'stoploss': 'market', 'stoploss_on_exchange': False}
2023-07-27 19:26:39,041 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using order_time_in_force: {'entry': 'GTC', 'exit': 'GTC'}
2023-07-27 19:26:39,041 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stake_currency: EUR
2023-07-27 19:26:39,041 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stake_amount: unlimited
2023-07-27 19:26:39,041 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using protections: []
2023-07-27 19:26:39,042 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using startup_candle_count: 60
2023-07-27 19:26:39,042 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using unfilledtimeout: {'entry': 10, 'exit': 10, 'exit_timeout_count': 0, 'unit': 'minutes'}
2023-07-27 19:26:39,042 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using use_exit_signal: True
2023-07-27 19:26:39,042 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using exit_profit_only: False
2023-07-27 19:26:39,042 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using ignore_roi_if_entry_signal: False
2023-07-27 19:26:39,042 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using exit_profit_offset: 0.0
2023-07-27 19:26:39,043 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using disable_dataframe_checks: False
2023-07-27 19:26:39,043 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using ignore_buying_expired_candle_after: 0
2023-07-27 19:26:39,043 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using position_adjustment_enable: False
2023-07-27 19:26:39,043 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using max_entry_position_adjustment: -1
2023-07-27 19:26:39,043 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using max_open_trades: 3
2023-07-27 19:26:39,043 - freqtrade.configuration.config_validation - INFO - Validating configuration ...
2023-07-27 19:26:39,098 - freqtrade.resolvers.iresolver - INFO - Using resolved pairlist StaticPairList from '/freqtrade/freqtrade/plugins/pairlist/StaticPairList.py'...
2023-07-27 19:26:39,250 - freqtrade.data.history.history_utils - INFO - Using indicator startup period: 60 ...
2023-07-27 19:26:39,628 - freqtrade.data.history.idatahandler - WARNING - No history for SOL/EUR, spot, 1h found. Use `freqtrade download-data` to download the data
2023-07-27 19:26:39,703 - freqtrade.optimize.backtesting - INFO - Loading data from 2022-12-18 00:00:00 up to 2023-06-16 06:00:00 (180 days).
2023-07-27 19:26:39,703 - freqtrade.configuration.timerange - WARNING - Moving start-date by 60 candles to account for startup time.
2023-07-27 19:26:39,704 - freqtrade.optimize.backtesting - INFO - Dataload complete. Calculating indicators
2023-07-27 19:26:39,705 - freqtrade.optimize.backtesting - WARNING - Backtest result caching disabled due to use of open-ended timerange.
2023-07-27 19:26:39,705 - freqtrade.optimize.backtesting - INFO - Running backtesting for Strategy SmallStraV3
2023-07-27 19:26:39,706 - freqtrade.strategy.hyper - INFO - No params for buy found, using default values.
2023-07-27 19:26:39,706 - freqtrade.strategy.hyper - INFO - Strategy Parameter(default): buy_rsi = 30
2023-07-27 19:26:39,707 - freqtrade.strategy.hyper - INFO - No params for sell found, using default values.
2023-07-27 19:26:39,708 - freqtrade.strategy.hyper - INFO - Strategy Parameter(default): sell_rsi = 70
2023-07-27 19:26:39,708 - freqtrade.strategy.hyper - INFO - No params for protection found, using default values.
2023-07-27 19:26:40,272 - freqtrade.optimize.backtesting - INFO - Backtesting with data from 2022-12-20 12:00:00 up to 2023-06-16 06:00:00 (177 days).
2023-07-27 19:26:50,433 - freqtrade.misc - INFO - dumping json to "/freqtrade/user_data/backtest_results/backtest-result-2023-07-27_19-26-50.meta.json"
2023-07-27 19:26:50,434 - freqtrade.misc - INFO - dumping json to "/freqtrade/user_data/backtest_results/backtest-result-2023-07-27_19-26-50.json"
2023-07-27 19:26:50,449 - freqtrade.misc - INFO - dumping json to "/freqtrade/user_data/backtest_results/.last_result.json"

```

**Comment by xmatthias** [comment]
You've had me scratch my head for a second there - but it's actually an error / mistake in your code that's causing this.


I've added inline comments below to highlight where the problem happens

```
    def informative_pairs(self):
        pairs = self.dp.current_whitelist()
        # Assign tf to each pair so they can be downloaded and cached for strategy.
        informative_pairs = [(pair, '4h') for pair in pairs]
        return informative_pairs
  # The above is necessary for live, has no impact on backtesting / the explained below


    def populate_indicators(self, dataframe):
        inf_tf = '4h'
        informative = self.dp.get_pair_dataframe(pair=metadata['pair'], timeframe=inf_tf)
        # Until here - all fine - make sure to not forget adding the combinations to informative_pairs

        informative['date'] = dataframe['date']          # informative contains 4h candles - in my test dataframe was 1h candles
        informative['close'] = dataframe['close']        # I'm not sure why you would do that - it'll overwrite the informative pair's close price - manipulating the 4h candles - and with invalid data at that.

        # The rest is fine again
        informative['rsi'] = ta.RSI(informative)
        infoMacD = ta.MACD(informative)
        informative['macd'] = infoMacD['macd']
        informative['macdsignal'] = infoMacD['macdsignal']
        informative['macdhist'] = infoMacD['macdhist']
        # informative['macd'] = macd['macd'](informative, timeperiod=14)
        dataframe = merge_informative_pair(dataframe, informative, self.timeframe, inf_tf, ffill=True) 
```

The 2 "problematic" assignments above assign a lower timeframe (say, 1h) to a 4h dataframe

so 
```
400 2023-02-28 08:00:00+00:00  1619.23  1631.87  1616.34  1629.53   46295.2908
401 2023-02-28 12:00:00+00:00  1629.52  1647.52  1628.81  1641.30   85830.8307
402 2023-02-28 16:00:00+00:00  1641.29  1646.38  1622.46  1628.01   65272.7667
403 2023-02-28 20:00:00+00:00  1628.01  1633.35  1596.66  1605.23  109543.9413
404 2023-03-01 00:00:00+00:00  1605.24  1640.72  1595.00  1638.38   97924.7560
```

becomes 

```
                         date     open     high      low    close       volume
400 2023-01-15 14:00:00+00:00  1619.23  1631.87  1616.34  1540.09   46295.2908
401 2023-01-15 15:00:00+00:00  1629.52  1647.52  1628.81  1556.35   85830.8307
402 2023-01-15 16:00:00+00:00  1641.29  1646.38  1622.46  1545.25   65272.7667
403 2023-01-15 17:00:00+00:00  1628.01  1633.35  1596.66  1551.36  109543.9413
404 2023-01-15 18:00:00+00:00  1605.24  1640.72  1595.00  1547.87   97924.7560
```

While the values didn't change - the dates did - so now you've suddenly got a 1h dataframe (the candles are still 4h), which is ending WAY early (a 4h dataframe is way shorter than a 1h dataframe).

Now all further lines act on the assumed 1h data - also the merge - which obviously will then end early - and will throw away all but every 4th candle.

simply put - remove the 2 faulty assignments - and this will no longer happen.

**Comment by M-t-t-h** [comment]
![Screenshot_3](https://github.com/freqtrade/freqtrade/assets/8056358/f2b4ed96-a6e6-4890-8bc6-d620f78110d1)

Thanks a lot !!

To be totaly honest, i added close and date cause on create containers errors said me it was missing...
and i craft them (in bad way...)

I missed 
```
 def informative_pairs(self):
        pairs = self.dp.current_whitelist()
        # Assign tf to each pair so they can be downloaded and cached for strategy.
        informative_pairs = [(pair, '4h') for pair in pairs]
        return informative_pairs
```

Thanks for time and explainations !
